### PR TITLE
[Merged by Bors] - feat(RepresentationTheory/Action): add preservation of (co)limits lemmas for functors into `Action V G`

### PR DESCRIPTION
**Pull request by chrisflav**
Shows that a functor into `Action V G` preserves a given (co)limit if it does after postcomposing with the forgetful functor to `V`. 

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RepresentationTheory/Action/Limits.lean [hunk 1]
```diff
@@ -53,6 +53,96 @@ instance [HasColimits V] : HasColimits (Action V G) :=
...
+/-- `F : C ⥤ Action V G` preserves the limit of some `K : J ⥤ C` if
+if it does after postcomposing with the forgetful functor `Action V G ⥤ V`. -/
+def preservesLimitOfPreserves (F : C ⥤ Action V G) {J : Type w₁}
+    [Category.{w₂, w₁} J] (K : J ⥤ C)
+    (h : PreservesLimit K (F ⋙ Action.forget V G)) : PreservesLimit K F := by
+  let F' : C ⥤ SingleObj G ⥤ V := F ⋙ (Action.functorCategoryEquivalence V G).functor
+  let i : PreservesLimit K F' := by
+    apply SingleObj.preservesLimit
+    show PreservesLimit K (F ⋙ Action.forget V G)
+    assumption
```
Comment (joelriou) [inline Mathlib/RepresentationTheory/Action/Limits.lean:79]:
```suggestion
    exact h
```

File: Mathlib/RepresentationTheory/Action/Limits.lean [hunk 2]
```diff
@@ -53,6 +53,96 @@ instance [HasColimits V] : HasColimits (Action V G) :=
 
 end Limits
 
+section Preservation
+
+variable {C : Type t₁} [Category.{t₂, t₁} C]
```
Comment (joelriou) [inline Mathlib/RepresentationTheory/Action/Limits.lean:58]: A general remark, the two universes of categories can be introduced as follows:
```suggestion
variable {C : Type t₁} [Category.{t₂} C]
```

File: Mathlib/RepresentationTheory/Action/Limits.lean [hunk 3]
```diff
@@ -53,6 +53,96 @@ instance [HasColimits V] : HasColimits (Action V G) :=
...
+/-- `F : C ⥤ Action V G` preserves the colimit of some `K : J ⥤ C` if
+if it does after postcomposing with the forgetful functor `Action V G ⥤ V`. -/
+def preservesColimitOfPreserves (F : C ⥤ Action V G) {J : Type w₁}
+    [Category.{w₂, w₁} J] (K : J ⥤ C)
+    (h : PreservesColimit K (F ⋙ Action.forget V G)) : PreservesColimit K F := by
+  let F' : C ⥤ SingleObj G ⥤ V := F ⋙ (Action.functorCategoryEquivalence V G).functor
+  let i : PreservesColimit K F' := by
+    apply SingleObj.preservesColimit
+    show PreservesColimit K (F ⋙ Action.forget V G)
+    assumption
```
Comment (joelriou) [inline Mathlib/RepresentationTheory/Action/Limits.lean:121]:
```suggestion
    exact h
```

**Review by joelriou — APPROVED** [review]

**Comment by joelriou** [comment]
Thanks! 

bors merge

**Comment by digama0** [comment]
bors merge

**Pull request closed without merging**